blog/tidy-finance-dummy-data/index.qmd

```diff
@@ -0,0 +1,422 @@
+
+We deliberately use the *dummy* label because the data is not meaningful in the sense that it allows readers to actually replicate the results of the book. For legal reasons, the data does *not* contain any samples of the original data. We merely generate random numbers for all columns of the tables that we use throughout the books. 
+
+To generate the dummy database, we use the following packages:
+
+```{r}
+#| message: false
+library(tidyverse)
+library(RSQLite)
+```
+
+We use the original database as an input and initialize a dummy version where we later store the output tables. 
+
+```{r}
+tidy_finance <- dbConnect(
+  SQLite(),
+  "../../data/tidy_finance.sqlite",
+  extended_types = TRUE
+)
+
+tidy_finance_dummy <- dbConnect(
+  SQLite(),
+  "../../data/tidy_finance_dummy.sqlite",
+  extended_types = TRUE
+)
+```
+
+Since we draw random numbers for most of the columns, we also define a seed to ensure that the generated numbers are replicable. We also initialize vectors of dates of different frequencies over 10 years that we then use to create yearly, monthly, and daily data, respectively. 
+
+```{r}
+set.seed(1234)
+
+start_date <- as.Date("2003-01-01")
+end_date <- as.Date("2022-12-31")
+
+time_series_years <- seq(year(start_date), year(end_date), 1)
+time_series_months <- seq(start_date, end_date, "1 month")
+time_series_days <- seq(start_date, end_date, "1 day")
+```
+
+## Create macro dummy data
+
+We start by creating dummy data for the macroeconomic tables. The following code does this by replacing the original data with random numbers while preserving the time structure (either daily or monthly) and column names. We iterate over each table name in the `macro_tables` vector. For each table, we fetch the data from the database and store it in a `data_original` table. The code checks the column names of `data_original` to determine if the data is structured by months or by days. Depending on the structure, it sets the appropriate time series and date column name. For each relevant column (i.e., excluding columns with dates), the code generates a command to replace its values with random numbers drawn from a uniform distribution between 0 and 1. We then use the `!!!` operator to unlist and execute the list of commands. This trick actually helps us to avoid typing the same function for each column individually. Finally, each table with dummy data is written to the new database with dummy data.
+
+Note that we do not put any meaningful structure on the macro tables because they can be freely downloaded from the original sources - check out [Accessing and Managing Financial Data](../../r/accessing-and-managing-financial-data.qmd) to replace the dummy data with the actual data.
```

Comment:
I would actually not reproduce these tables. You could also just link the chapter and say that these are publicly available. Bad because you already had all the work, but really not necessary from my point of view.

Reply:
I prefer to have all tables fully covered, otherwise you need to pull code from different sources just to execute some code.